### PR TITLE
Updating signed out page with new design and button

### PR DESCRIPTION
**Pull request by MIslam92**
### What is the context of this PR?
- Added link to button to redirect respondent back to the sign in page
- Updated page messages displayed
- Button redirects user to RRM sign-in page 

### How to review 
Start any survey and hit "Save and Complete Later" to check if content on page matches design:
https://trello.com/c/7VACxFOq/1016-clear-actions-to-exit-or-continue-a-survey-s
Also, that the "Sign In" button returns the respondent back to the RRM sign in page to allow them to either continue or start another survey

### CODE REVIEW

**Comment by MIslam92** [comment]
Will need Hamish to go over whenever his free due how the usage of button is being handled rather then the usual in eq-survey-runner

File: app/templates/signed-out.html [hunk 1]
```diff
@@ -5,11 +5,11 @@
 {% block main %}
 
   <div class="u-mb-s">
-    <h1 class="saturn">Your survey has been saved</h1>
+    <h1 class="saturn">Your survey has been saved. You are <br> now signed out.</h1>
```
Comment (jonnyshaw89) [inline app/templates/signed-out.html]: Why do we have an explicit break in this H1?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

**Comment by yoganandkunche** [comment]
I think it is worth having the sign url on session-expired.html

File: app/assets/styles/partials/components/_buttons.scss [hunk 2]
```diff
@@ -2,7 +2,7 @@
   background-color: $bg;
   color: $color;
   border: none;
-  padding: 0.75rem 2.5rem;
+  padding: 0.75rem 4.5rem;
```
Comment (hamishtaplin) [inline app/assets/styles/partials/components/_buttons.scss]: What's the reason for this change? It'll make all buttons bigger everywhere.

File: app/templates/session-expired.html [hunk 3]
```diff
@@ -6,11 +6,12 @@
...
 
 <div class="u-mb-s">
-    <h1 class="saturn" {{helpers.track('visible', 'Timeout', 'Timeout expired')}}>Your session has expired. You are now </br> signed out.</h1>
+    <h1 class="saturn" {{helpers.track('visible', 'Timeout', 'Timeout expired')}}>Your session has expired. You are
+    now </br> signed out.</h1>
 </div>
 
 {% block submit_button %}
-<a href="https://survey.ons.gov.uk/" target="_blank" class="btn btn-default">Sign In</a>
+<a href="https://survey.onsdigital.gov.uk/" target="_blank" class="btn btn-default">Sign In</a>
```
Comment (hamishtaplin) [inline app/templates/session-expired.html]: This is a link and so should look like one.

File: app/templates/signed-out.html [hunk 4]
```diff
@@ -9,7 +9,7 @@ <h1 class="saturn">Your survey has been saved. You are <br> now signed out.</h1>
   </div>
 
 {% block submit_button %}
-<a href="https://survey.ons.gov.uk/" target="_blank" class="btn btn-default">Sign In</a>
+<a href="https://survey.onsdigital.gov.uk/" target="_blank" class="btn btn-default">Sign In</a>
```
Comment (hamishtaplin) [inline app/templates/signed-out.html]: Links shouldn't be styled as buttons.

**Review by hamishtaplin — CHANGES_REQUESTED** [review]

File: app/templates/session-expired.html [hunk 5]
```diff
@@ -5,12 +5,13 @@
...
+<div class="u-mb-s">
+    <h1 class="saturn" {{helpers.track('visible', 'Timeout', 'Timeout expired')}}>Your session has expired. You are
+    now </br> signed out.</h1>
+</div>
 
-  <div class="panel panel--simple panel--success panel--spacious">
-    <p>To help protect your information we have signed you out.</p>
-  </div>
+{% block submit_button %}
+<a href="https://survey.onsdigital.gov.uk/" target="_blank" class="btn btn-default">Sign In</a>
```
Comment (collisdigital) [inline app/templates/session-expired.html]: URL won't work in prod?

File: app/templates/signed-out.html [hunk 6]
```diff
@@ -5,11 +5,11 @@
...
   <div class="u-mb-s">
-    <h1 class="saturn">Your survey has been saved</h1>
+    <h1 class="saturn">Your survey has been saved. You are <br> now signed out.</h1>
   </div>
 
-  <div class="panel panel--simple panel--success panel--spacious">
-    <p>You have been signed out of the survey, to complete the survey you will need to sign back in.</p>
-  </div>
+{% block submit_button %}
+<a href="https://survey.onsdigital.gov.uk/" target="_blank" class="btn btn-default">Sign In</a>
```
Comment (collisdigital) [inline app/templates/signed-out.html]: Should this be duplicated here as already in `session-expired.html`

File: app/templates/signed-out.html (hunk 1 above)
Comment (MIslam92) [reply app/templates/signed-out.html]: To match design on screenshots on the card i.e. having the text on two lines: https://trello.com/c/7VACxFOq/1016-clear-actions-to-exit-or-continue-a-survey-s

Comment (jonnyshaw89) [reply app/templates/signed-out.html]: I think we should be relying more on the browser wrapping the text rather then putting explicit breaks in?

Comment (hamishtaplin) [reply app/templates/signed-out.html]: @jonnyshaw89 @MIslam92  I agree

File: app/templates/session-expired.html [hunk 7]
```diff
@@ -5,12 +5,16 @@
...
+<div class="u-mb-s">
     <h1 class="saturn" {{helpers.track('visible', 'Timeout', 'Timeout expired')}}>Your session has expired</h1>
-  </div>
+</div>
 
   <div class="panel panel--simple panel--success panel--spacious">
     <p>To help protect your information we have signed you out.</p>
   </div>
 
+<br>
```
Comment (hamishtaplin) [inline app/templates/session-expired.html]: Use a margin utility class (eg `u-mb-s`) on the `panel` instead of a `<br>`. Only use `<br>` for line-breaks in text, not to visually separate content.

**Review by hamishtaplin — CHANGES_REQUESTED** [review]

File: app/templates/signed-out.html (hunk 1 above)
Comment (hamishtaplin) [reply app/templates/signed-out.html]: Just to illustrate the point of why you should be judicious about using line-breaks:

screenshot 2017-10-05 13 31 57

File: app/templates/session-expired.html [hunk 8]
```diff
@@ -5,12 +5,16 @@
...
+
+<a href="https://survey.onsdigital.gov.uk/" target="_blank">Return to your account</a>
```
Comment (hamishtaplin) [inline app/templates/session-expired.html]: Is there a reason to do this in a new tab (`target="_blank"`)?

**Review by hamishtaplin — CHANGES_REQUESTED** [review]

File: app/templates/session-expired.html [hunk 9]
```diff
@@ -5,12 +5,17 @@
...
     <h1 class="saturn" {{helpers.track('visible', 'Timeout', 'Timeout expired')}}>Your session has expired</h1>
-  </div>
+</div>
 
-  <div class="panel panel--simple panel--success panel--spacious">
+  <div class="panel panel--simple panel--success panel--spacious u-mb-s">
     <p>To help protect your information we have signed you out.</p>
   </div>
 
+{% if env == EQ_PREPROD_ENV %}
```
Comment (shahi645) [inline app/templates/session-expired.html]: How does the app know what environment it is in ?

File: app/settings.py [hunk 10]
```diff
@@ -63,7 +63,11 @@ def get_env_or_fail(key):
 EQ_SERVER_SIDE_STORAGE_DATABASE_HOST = os.getenv('EQ_SERVER_SIDE_STORAGE_DATABASE_HOST')
 EQ_SERVER_SIDE_STORAGE_DATABASE_PORT = os.getenv('EQ_SERVER_SIDE_STORAGE_DATABASE_PORT', 5432)
 EQ_SERVER_SIDE_STORAGE_DATABASE_NAME = os.getenv('EQ_SERVER_SIDE_STORAGE_DATABASE_NAME', 'digitaleqrds')
-EQ_SERVER_SIDE_STORAGE_USER_ID_ITERATIONS = ensure_min(os.getenv('EQ_SERVER_SIDE_STORAGE_USER_ID_ITERATIONS', 10000), 1000)
+EQ_SERVER_SIDE_STORAGE_USER_ID_ITERATIONS = ensure_min(os.getenv('EQ_SERVER_SIDE_STORAGE_USER_ID_ITERATIONS', 10000),
+                                                       1000)
+
+EQ_PREPROD_ENV = os.getenv("https://surveypreprod.ons.gov.uk/")
+EQ_PROD_ENV = os.getenv("https://survey.onsdigital.gov.uk/")
```
Comment (shahi645) [inline app/settings.py]: These need to be set to eq preprod and prod? not rrm ?

Comment (jonnyshaw89) [reply app/settings.py]: This should be a variable of something like UPSTREAM_LOGIN_PAGE and not per environment

File: app/templates/session-expired.html (hunk 9 above)
Comment (jonnyshaw89) [reply app/templates/session-expired.html]: The URL should be pulled from the variable. not using the variable to select a hard coded URL

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: app/storage/metadata_parser.py [hunk 11]
```diff
@@ -42,25 +42,26 @@ def generate(self):
...
+    "survey_url": MetadataField(mandatory=False),
+    "return_by": MetadataField(validator=iso_8601_data_parser),
+    "trad_as": MetadataField(mandatory=False),
+    "employment_date": MetadataField(mandatory=False, validator=iso_8601_data_parser),
+    "region_code": MetadataField(mandatory=False),
+    "language_code": MetadataField(mandatory=False),
+    "tx_id": MetadataField(mandatory=False, validator=uuid_4_parser, generator=id_generator),
+    "variant_flags": MetadataField(mandatory=False),
+    "roles": MetadataField(mandatory=False),
+    "return_URL": MetadataField(mandatory=False),
```
Comment (jonnyshaw89) [inline app/storage/metadata_parser.py]: this should be lower case to match the convention

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: app/settings.py [hunk 12]
```diff
@@ -63,11 +63,15 @@ def get_env_or_fail(key):
 EQ_SERVER_SIDE_STORAGE_DATABASE_HOST = os.getenv('EQ_SERVER_SIDE_STORAGE_DATABASE_HOST')
 EQ_SERVER_SIDE_STORAGE_DATABASE_PORT = os.getenv('EQ_SERVER_SIDE_STORAGE_DATABASE_PORT', 5432)
 EQ_SERVER_SIDE_STORAGE_DATABASE_NAME = os.getenv('EQ_SERVER_SIDE_STORAGE_DATABASE_NAME', 'digitaleqrds')
-EQ_SERVER_SIDE_STORAGE_USER_ID_ITERATIONS = ensure_min(os.getenv('EQ_SERVER_SIDE_STORAGE_USER_ID_ITERATIONS', 10000), 1000)
+EQ_SERVER_SIDE_STORAGE_USER_ID_ITERATIONS = ensure_min(os.getenv('EQ_SERVER_SIDE_STORAGE_USER_ID_ITERATIONS', 10000),
+                                                       1000)
+
+RESPONDENT_ACCOUNT_URL = os.getenv('RESPONDENT_ACCOUNT_URL', 'https://survey.onsdigital.gov.uk/')
```
Comment (jonnyshaw89) [inline app/settings.py]: Generally we default to production values.

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: app/templates/macros/helpers.html [hunk 13]
```diff
@@ -28,3 +28,18 @@
...
+<a href="https://surveypreprod.ons.gov.uk/" {{attr|xmlattr}}>Return to your account</a>
+{%- endmacro -%}
+
+{%- macro prod(class=None) -%}
+  {% set attr = {
+    "class": class
+  } %}
+<a href="https://survey.onsdigital.gov.uk/" {{attr|xmlattr}}>Return to your account</a>
+{%- endmacro -%}
+
```
Comment (shahi645) [inline app/templates/macros/helpers.html]: We are not using these anymore, so can be removed?

File: app/views/errors.py [hunk 14]
```diff
@@ -104,4 +104,5 @@ def render_template(template_name):
                                  analytics_ua_id=current_app.config['EQ_UA_ID'],
                                  ua=user_agent,
                                  tx_id=tx_id,
+                                 respondent_account_url=current_app.config['RESPONDENT_ACCOUNT_URL'],
```
Comment (shahi645) [inline app/views/errors.py:107]: Logically, this shouldn't be in the errors views

File: app/storage/metadata_parser.py [hunk 15]
```diff
@@ -61,6 +61,7 @@ def generate(self):
     'tx_id': MetadataField(mandatory=False, validator=uuid_4_parser, generator=id_generator),
     'variant_flags': MetadataField(mandatory=False),
     'roles': MetadataField(mandatory=False),
+    'return_url': MetadataField(mandatory=False),
```
Comment (jonnyshaw89) [inline app/storage/metadata_parser.py]: Is this not used?

File: tests/integration/views/test_session.py [hunk 16]
```diff
@@ -9,4 +9,4 @@ def test_session_expired(self):
 
     def test_session_signed_out(self):
         self.get('/signed-out')
-        self.assertInPage('Your survey has been saved')
+        self.assertInPage('Your survey answers have been saved')
```
Comment (shahi645) [inline tests/integration/views/test_session.py:12]: Should we also add a test that checks the link goes to RRM/RAS prod/preprod?

Comment (jonnyshaw89) [reply tests/integration/views/test_session.py:12]: by asserting the href and not by clicking it

File: tests/integration/views/test_session.py [hunk 17]
```diff
@@ -1,4 +1,5 @@
 from tests.integration.integration_test_case import IntegrationTestCase
+from app.settings import RESPONDENT_ACCOUNT_URL
```
Comment (jonnyshaw89) [inline tests/integration/views/test_session.py:2]: I don't think this is used?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: app/templates/signed-out.html [hunk 18]
```diff
@@ -4,12 +4,11 @@
...
-  </div>
 
-  <div class="panel panel--simple panel--success panel--spacious">
-    <p>You have been signed out of the survey, to complete the survey you will need to sign back in.</p>
-  </div>
+<div class="u-mb-s">
+    <h1 class="saturn">Your survey answers have been saved. You are now signed out.</h1>
+</div>
+
+<a href={{respondent_account_url}}>Return to your account</a>
```
Comment (jonnyshaw89) [inline app/templates/signed-out.html]: When i run this it renders as `<a href=https://survey.ons.gov.uk/>Return to your account</a>`
should the url have quotes around it?

File: app/templates/signed-out.html [hunk 19]
```diff
@@ -4,12 +4,11 @@
...
-  </div>
 
-  <div class="panel panel--simple panel--success panel--spacious">
-    <p>You have been signed out of the survey, to complete the survey you will need to sign back in.</p>
-  </div>
+<div class="u-mb-s">
+    <h1 class="saturn">Your survey answers have been saved. You are now signed out.</h1>
+</div>
+
+<a href='{{respondent_account_url}}'>Return to your account</a>
```
Comment (jonnyshaw89) [inline app/templates/signed-out.html]: Double quotes

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: tests/integration/views/test_session.py [hunk 20]
```diff
@@ -9,3 +10,7 @@ def test_session_expired(self):
     def test_session_signed_out(self):
         self.get('/signed-out')
         self.assertInPage('Your survey answers have been saved')
+
+    def test_rrm_page(self):
```
Comment (jonnyshaw89) [inline tests/integration/views/test_session.py]: This names doesn't make sense. the assert below should just be moved into `test_session_signed_out`

File: tests/integration/views/test_session.py [hunk 21]
```diff
@@ -1,12 +1,16 @@
...
         self.assertInPage('Your session has expired')
 
     def test_session_signed_out(self):
         self.get('/signed-out')
-        self.assertInPage('Your survey has been saved')
+        self.assertInPage('Your survey answers have been saved')
+
+    def test_rrm_page(self):
+        self.get('/signed-out')
+        self.assertInPage(RESPONDENT_ACCOUNT_URL)
```
Comment (jonnyshaw89) [inline tests/integration/views/test_session.py]: Move this line into `test_session_signed_out` and delete `test_rrm_page`

**Review by jonnyshaw89 — APPROVED** [review]

**Review by hamishtaplin — APPROVED** [review]
